src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -246,6 +246,11 @@ void java_bytecode_convert_classt::convert(
 /// java::array[X] where X is byte, float, int, char...
 void java_bytecode_convert_classt::add_array_types()
 {
+  // this method does not need to be called more than once
+  static bool already_executed=false;
+  if(already_executed)
+    return;
+
```

Comment:
The previous version was calling this method once for each class being converted to `codet`, which is even worse.

If you are really against this fix I can take it out and create an issue. Let me know.